pymeasure/instruments/agilent/agilentL4534A.py

```diff
@@ -0,0 +1,333 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import numpy as np
+from numpy import typing
+
+from decimal import Decimal
+
+from pymeasure.errors import Error
+from pymeasure.instruments import Instrument, Channel
+from pymeasure.instruments.validators import strict_discrete_set, strict_range, strict_discrete_range
+from pymeasure.units import ureg
+
+
+def sample_count_function(value, values): return strict_discrete_range(value, values, 4)
+
+SAMPLE_COUNT_VALUES = [8, 128e6]
+PRE_TRIG_VALUES = [0, 128e6]
+RECORD_COUNT_VALUES = [1, 1024]
+
+TRIGGER_HOLDOFF_VALUES = ureg.Quantity(np.asarray([0, 10]), ureg.s)
+TRIGGER_DELAY_VALUES = ureg.Quantity([0, 3600], ureg.s)
+
+def _get_channel_config_process(values):
+    result = {
+        'range': ureg.Quantity(Decimal(values[0]), ureg.V),
+        'coupling': values[1],
+        'filter': values[2]
+    }
+    return result
+
+def _get_acq_config_process(values):
+    return {
+        'sample_rate': ureg.Quantity(values[0], ureg.Hz),
+        'samples_per_record': int(values[1]),
+        'pre_trig_samples': int(values[2]),
+        'num_records': int(values[3]),
+        'trigger_holdoff': ureg.Quantity(values[4], ureg.s),
+        'trigger_delay': ureg.Quantity(values[5], ureg.s)
+    }
+
+
+def _set_acq_config_process(value):
+    return '{},{},{},{},{},{}'.format(
+        value['sample_rate'].m_as(ureg.Hz),
+        value['samples_per_record'],
+        value['pre_trig_samples'],
+        value['num_records'],
+        value['trigger_holdoff'].m_as(ureg.s),
+        value['trigger_delay'].m_as(ureg.s))
+
+
+VOLTAGE_RANGE_VALUES = ureg.Quantity(np.asarray([0.25, 0.5, 1, 2, 4, 8, 16, 32, 128, 256]), ureg.V)
+
+COUPLING_VALUES = ['DC', 'AC']
+FILTER_VALUES = ['LP_20_MHZ', 'LP_2_MHZ', 'LP_200_KHZ']
+    
+class AgilentL4534A(Instrument):
+    """
+    Represents the Agilent L4532A/L4534A digitizers.
+    """
+
+    SAMPLE_RATE_VALUES = ureg.Quantity(np.asarray([1000, 2000, 5000, 10000, 20000, 50000,
+                                    100000, 200000, 500000, 1000000, 2000000, 5000000, 10000000, 20000000]), ureg.Hz)
+
+    @staticmethod
+    def _validate_acq_config(value, values):
+        strict_discrete_set(value['sample_rate'], AgilentL4534A.SAMPLE_RATE_VALUES)
+        strict_discrete_range(value['samples_per_record'], SAMPLE_COUNT_VALUES, 4)
+        strict_discrete_range(value['pre_trig_samples'], PRE_TRIG_VALUES, 4)
+        strict_range(value['num_records'], RECORD_COUNT_VALUES)
+        strict_range(value['trigger_holdoff'], TRIGGER_HOLDOFF_VALUES)
+        strict_range(value['trigger_delay'], TRIGGER_DELAY_VALUES)
+        return value
+
+    class Channel(Channel):
```

Comment:
I think it is better not to replace the (pymeasure) Channel with your new Channel, but to use a new name.